qutebrowser/mainwindow/tabwidget.py

```diff
@@ -447,38 +453,48 @@ def minimumTabSizeHint(self, index, ellipsis: bool = True):
         Return:
             A QSize of the smallest tab size we can make.
         """
-        text = '\u2026' if ellipsis else self.tabText(index)
+        icon = self.tabIcon(index)
+        extent = self.style().pixelMetric(QStyle.PM_TabBarIconSize, None, self)
+        if not icon:
+            icon_width = 0
+        else:
+            icon_width = icon.actualSize(QSize(extent, extent)).width()
+        return self._minimum_tab_size_hint_helper(self.tabText(index),
+                                                  icon_width,
+                                                  ellipsis)
+
+    @functools.lru_cache(maxsize=2**10)
```

Comment:
Per the [functools docs](https://docs.python.org/3/library/functools.html), they reccomend maxsizes as powers of two, so I made it one. I think that 2^10 is a good value, I'm not sure if there's a better way to verify this though (or scaling it with the number of tabs).

Reply:
What I do for other caches is adding them to `debug_cache_stats` in `qutebrowser/misc/utilcmds.py`. Then you can get some statistics on stdout by calling that.

This might be a bit more difficult here because there are multiple windows, but I still think it'd be a good idea.

Reply:
Playing with it a bit, I get `tab width cache: CacheInfo(hits=51981, misses=111, maxsize=1024, currsize=111)`

So I think 1024 is fine, maybe we could lower it to 512 even.